### PR TITLE
Address Printf formatting issues

### PR DESCRIPTION
**Pull request by pan-**
#### Problem
Not all unit tests can be built on 32 bits ARM targets due to printf formatting issue.

#### Change overview
Use C99 formatters to print correctly on all architecture.

#### Testing
How was this tested? (at least one bullet point required)
=> Compiled and run with Mbed OS unit test (PR to follow).

### CODE REVIEW

File: src/inet/tests/TestInetCommonPosix.cpp [hunk 1]
```diff
@@ -450,7 +450,11 @@ void ServiceEvents(struct ::timeval & aSleepTime)
         if (NetworkIsReady())
 #endif
         {
+#ifdef __MBED__
```
Comment (andy31415) [inline src/inet/tests/TestInetCommonPosix.cpp:453]: Could we just not write pid data for any of the variants?
I am unsure if it is used and it seems like debug code only. I would prefer less `#ifdef`

**Review by andy31415 — APPROVED** [review]

File: src/inet/tests/TestInetCommonPosix.cpp (hunk 1 above)
Comment (pan-) [reply src/inet/tests/TestInetCommonPosix.cpp:453]: Yes, we were surprised of having these prints. It's not like `getpid` and `getppid` are very common on embedded targets.

Comment (pan-) [reply src/inet/tests/TestInetCommonPosix.cpp:453]: Yes, we were surprised of having these prints. It's not like `getpid` and `getppid` are very common on embedded targets.

**Review by jmartinez-silabs — APPROVED** [review]

File: src/inet/tests/TestInetCommonPosix.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply src/inet/tests/TestInetCommonPosix.cpp:453]: I'd be in favor of just not printing pid/ppid here.

File: src/inet/tests/TestInetErrorStr.cpp [hunk 2]
```diff
@@ -76,7 +76,7 @@ static void CheckInetErrorStr(nlTestSuite * inSuite, void * inContext)
     Inet::RegisterLayerErrorFormatter();
 
     // For each defined error...
-    for (int err : sContext)
+    for (int32_t err : sContext)
```
Comment (bzbarsky-apple) [inline src/inet/tests/TestInetErrorStr.cpp:79]: Just as a warning, this will conflict with #8383...  Just something to watch out for.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/inet/tests/TestInetCommonPosix.cpp (hunk 1 above)
Comment (pan-) [reply src/inet/tests/TestInetCommonPosix.cpp]: I removed the printing.

**Pull request merged**